### PR TITLE
.github: run Sonar Check only on PRs from apache/cloudstack branches

### LINKED ISSUE #8050: Main Branch Sonar Quality Check failing
Main Branch Sonar Quality Check / Main Sonar JaCoCo Build is failing Github Actions on main and likely PRs for quite sometime. We must look at how to fix it, or remove it if the integration isn't possible.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Github Actions
~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #8050 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: .github/workflows/sonar-check.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ concurrency:
 
 jobs:
   build:
-    if: github.repository == 'apache/cloudstack'
+    if: github.repository == 'apache/cloudstack' && github.event.pull_request.head.repo.full_name == github.repository
```
Comment (DaanHoogland) [inline .github/workflows/sonar-check.yml:31]:
```suggestion
    if: github.event.pull_request.head.repo.full_name == github.repository
```
i.e. people may run sonar on their own forks?

Comment (weizhouapache) [reply .github/workflows/sonar-check.yml:31]: if so, users need to configure sonar token in github project.

For example, if someone creates a PR in shapeblue (forked) repositories, currently the sonar check is skipped. 
But with this change, sonar check will not be skipped, but fail (as sonar token does not exist in project settings).

Comment (DaanHoogland) [reply .github/workflows/sonar-check.yml:31]: but only if they create a PR from their own repo to their own repo, so they have control on whether they create the token or not.

Comment (weizhouapache) [reply .github/workflows/sonar-check.yml:31]: yes. so between these 2 options with forks
- enable PR sonar check by default. users need to disable it by code changes if they do not have sonar token.
- disable PR sonar check by default. users need to enable it by code changes if they have sonar token.

the PR sonar check was disabled on forks by #7199 
we can change the behaviour if needed

Comment (DaanHoogland) [reply .github/workflows/sonar-check.yml:31]: but in both cases the code may not be merged, inconvenient :(

Comment (weizhouapache) [reply .github/workflows/sonar-check.yml:31]: Right...
Let's assume that most users do not have sonar token, and run sonar check only on apache/cloudstack repository. Make sense ?

**Review by vishesh92 — APPROVED** [review]
clgtm

**Review by rohityadavcloud — APPROVED** [review]

**Review by shwstppr — APPROVED** [review]
LGTM

**Review by DaanHoogland — APPROVED** [review]

**Pull request merged**